appmap/_implementation/generation.py

```diff
@@ -1,67 +1,48 @@
 """Generate an AppMap"""
-import json
-from collections.abc import MutableMapping
+import orjson
+from dataclasses import dataclass, field
 
 from . import metadata
+from .event import Event
 
-from .event import Event, serialize_event
 
-
-class ClassMapSet(MutableMapping):
-    def __init__(self):
-        self.dict = dict()
-
-    def __delitem__(self, k):
-        del self[k]
-
-    def __getitem__(self, k):
-        return self.dict[k]
-
-    def __setitem__(self, k, v):
-        self.dict[k] = v
-
-    def __iter__(self):
-        return self.dict.__iter__()
-
-    def __len__(self):
-        return len(self.dict)
+class ClassMapDict(dict):
+    pass
 
 
+@dataclass
 class ClassMapEntry:
-    def __init__(self, key, name, entry_type):
-        self.key = key
-        self.name = name
-        self.type = entry_type
-
-    def __eq__(self, other):
-        return self.key == other.key
-
-    def __hash__(self):
-        return hash(self.key)
+    name: str
+    type: str
```

Comment:
@apotterri `type` is a system variable

Reply:
How much of a problem is this? Python isn't currently objecting to it, and the AppMap attribute needs to be named `type`. I'm sure it can be changed if absolutely necessary....

Reply:
nah, scratch that. its an attribute not variable, so it should be fine.